packages/firebase/index.d.ts

```diff
@@ -8310,10 +8312,21 @@ declare namespace firebase.firestore {
      */
     set<T>(
       documentRef: DocumentReference<T>,
-      data: T,
-      options?: SetOptions
+      data: Partial<T>,
+      options: SetOptions
     ): Transaction;
 
+    /**
+     * Writes to the document referred to by the provided `DocumentReference`.
+     * If the document does not exist yet, it will be created. If you pass
+     * `SetOptions`, the provided data can be merged into the existing document.
+     *
+     * @param documentRef A reference to the document to be set.
+     * @param data An object of the fields and values for the document.
```

Comment:
Sorry if I'm misreading this, but is SetOptions now required? Do we need a param for it?

Reply:
We are adding an overload that allows to specify a `data: Partial<T>` parameter that must be accompanied by `SetOptions`. Using the `data: T` overload does not require `SetOptions`.